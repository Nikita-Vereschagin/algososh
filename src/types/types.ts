import { ElementStates } from "./element-states";

export type TSwap = {letter?: string , number: number, state: ElementStates}

export type TStack = {
    push(el: string): void;
    pop(): void;
    clear(): void;
    getArray(): string[];
}

export type TQueue = {
    enqueue(el: string): void;
    dequeue(): void;
    clear(): void;
    getArray(): string[];
    getIndex(): {head: number, tail: number}
}

export type TList = {
    addHelper(i: number, el: string): void;
    delHelper(i: number): void;
    getArray(): string[];
}