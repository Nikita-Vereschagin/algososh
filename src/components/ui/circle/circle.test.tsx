

import renderer from 'react-test-renderer';
import { Circle } from "./circle";

import { ElementStates } from "../../../types/element-states";

it('Круг без буквы', () => {
    const tree = renderer
        .create(<Circle letter={undefined}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
}); 

it('Круг с буквой', () => {
    const tree = renderer
        .create(<Circle letter="A"/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
}); 

it('Круг с head', () => {
    const tree = renderer
        .create(<Circle head='1'/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
}); 

it('Круг с reactEl в head', () => {
    const tree = renderer
        .create(<Circle head={<Circle letter="B"/>}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
}); 

it('Круг с tail', () => {
    const tree = renderer
        .create(<Circle tail='2'/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
}); 

it('Круг с reactEl в tail', () => {
    const tree = renderer
        .create(<Circle tail={<Circle letter="C"/>}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
}); 

it('Круг с index', () => {
    const tree = renderer
        .create(<Circle index={99}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
}); 

it('Круг с isSmall', () => {
    const tree = renderer
        .create(<Circle isSmall={true}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
}); 

it('Круг в состояние Default', () => {
    const tree = renderer
        .create(<Circle state={ElementStates.Default}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
}); 

it('Круг в состояние Changing', () => {
    const tree = renderer
        .create(<Circle state={ElementStates.Changing}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
}); 

it('Круг в состояние Modified', () => {
    const tree = renderer
        .create(<Circle state={ElementStates.Modified}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
}); 



