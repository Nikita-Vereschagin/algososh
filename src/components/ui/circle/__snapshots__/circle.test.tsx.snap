// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Круг без буквы 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Круг в состояние Changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Круг в состояние Default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Круг в состояние Modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Круг с head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    1
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Круг с index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    99
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Круг с isSmall 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Круг с reactEl в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          B
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Круг с reactEl в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          C
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Круг с tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    2
  </div>
</div>
`;

exports[`Круг с буквой 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
